govcd/adminvdc_test.go

```diff
@@ -0,0 +1,239 @@
+		check.Skip(fmt.Sprintf("No Provider VDC found with name '%s'", vcd.config.VCD.ProviderVdc.Name))
+	}
+	providerVdcHref := results.Results.VMWProviderVdcRecord[0].HREF
+
+	results, err = vcd.client.QueryWithNotEncodedParams(nil, map[string]string{
+		"type":   "providerVdcStorageProfile",
+		"filter": fmt.Sprintf("name==%s", vcd.config.VCD.ProviderVdc.StorageProfile),
+	})
+	check.Assert(err, IsNil)
+	if len(results.Results.ProviderVdcStorageProfileRecord) == 0 {
+		check.Skip(fmt.Sprintf("No storage profile found with name '%s'", vcd.config.VCD.ProviderVdc.StorageProfile))
+	}
+	providerVdcStorageProfileHref := results.Results.ProviderVdcStorageProfileRecord[0].HREF
+
+	results, err = vcd.client.QueryWithNotEncodedParams(nil, map[string]string{
+		"type":   "networkPool",
+		"filter": fmt.Sprintf("name==%s", vcd.config.VCD.ProviderVdc.NetworkPool),
+	})
+	check.Assert(err, IsNil)
+	if len(results.Results.NetworkPoolRecord) == 0 {
+		check.Skip(fmt.Sprintf("No network pool found with name '%s'", vcd.config.VCD.ProviderVdc.NetworkPool))
+	}
+	networkPoolHref := results.Results.NetworkPoolRecord[0].HREF
+
+	allocationModels := []string{"AllocationVApp", "AllocationPool", "ReservationPool", "Flex"}
+	trueValue := true
+	for i, allocationModel := range allocationModels {
+		vdcConfiguration := &types.VdcConfiguration{
+			Name:            fmt.Sprintf("%s%d", TestCreateOrgVdc, i),
+			Xmlns:           types.XMLNamespaceVCloud,
+			AllocationModel: allocationModel,
+			ComputeCapacity: []*types.ComputeCapacity{
+				&types.ComputeCapacity{
+					CPU: &types.CapacityWithUsage{
+						Units:     "MHz",
+						Allocated: 1024,
+						Limit:     1024,
+					},
+					Memory: &types.CapacityWithUsage{
+						Allocated: 1024,
+						Limit:     1024,
+					},
+				},
+			},
+			VdcStorageProfile: []*types.VdcStorageProfile{&types.VdcStorageProfile{
+				Enabled: true,
+				Units:   "MB",
+				Limit:   1024,
+				Default: true,
+				ProviderVdcStorageProfile: &types.Reference{
+					HREF: providerVdcStorageProfileHref,
+				},
+			},
+			},
+			NetworkPoolReference: &types.Reference{
+				HREF: networkPoolHref,
+			},
+			ProviderVdcReference: &types.Reference{
+				HREF: providerVdcHref,
+			},
+			IsEnabled:            true,
+			IsThinProvision:      true,
+			UsesFastProvisioning: true,
+		}
+
+		if allocationModel == "Flex" {
+			vdcConfiguration.IsElastic = &trueValue
+			vdcConfiguration.IncludeMemoryOverhead = &trueValue
+		}
+
+		vdc, _ := adminOrg.GetVDCByName(vdcConfiguration.Name, false)
+		if vdc != nil {
+			err = vdc.DeleteWait(true, true)
+			check.Assert(err, IsNil)
+		}
+
+		// expecting fail due missing value
+		task, err := adminOrg.CreateOrgVdcAsync(vdcConfiguration)
+		check.Assert(err, Not(IsNil))
+		check.Assert(task, Equals, Task{})
+		// From vCD 10+ error changed
+		if vcd.client.Client.APIVCDMaxVersionIs("> 32.0") {
+			check.Assert(err.Error(), Equals, "error retrieving vdc: API Error: 400:  is not a valid unit. Please use MB")
```

Comment:
This comparison seems too strict. Can we just check that there is an API error 400? The message may change in future API versions

Reply:
changed